content/blog/2025-05-08-go-client-GA.md

```diff
@@ -0,0 +1,519 @@
+	}
+	defer clusterClient.Close()
+
+	// Create new transaction
+	clusterTx := pipeline.NewClusterBatch(true)
+
+	// Create ping options with specific routing
+	pingOpts := options.PingOptions{
+		Message: "Hello Valkey Cluster",
+	}
+
+	// Queue commands with routing
+	clusterTx.PingWithOptions(pingOpts)
+
+	// Command sequence
+	txs := []string{"Ping - 'Hello Valkey Cluster'"}
+
+	opts := pipeline.NewClusterBatchOptions().
+		WithRoute(config.RandomRoute)
+
+	// Execute transaction
+	results, err := clusterClient.ExecWithOptions(context.Background(), *clusterTx, false, *opts)
+	if err != nil {
+		log.Printf("Cluster transaction failed: %v", err)
+		return
+	}
+
+	// Process results
+	for i, result := range results {
+		fmt.Printf("Command %s results: %v\n", txs[i], result)
+	}
+```
+
+## Batch design
+
+The batch implementation in Valkey GLIDE was designed with several key goals, each achieved through specific implementation approaches:
+
+### Idiomatic Go API
+
+The design provides a natural, Go-like interface for batches through:
+
+- Clean command calling syntax (`tx.Set(...)`, `tx.Get(...)`)
+- Use of Go's embedding to inherit behavior from `BaseBatch`
+- Explicit error handling that follows Go conventions (`result, err := pipeline.Exec()`)
+- Use of method chaining to efficiently build command sequences (e.g. `tx.Set(...).Set(...).SAdd(...)`)
+- Use of Go context for timeout and cancellation support 
+
+### Command Transparency
+
+A important principle in the design of the client was transparently accessing the command, in both regular and batch contexts. While the server-side command execution is the same, the client-side result and error handling will differ since the batch results must account for multiple commands and potential errors. Go context is passed when the command is executed: with regular clients this occurs when the command is called, and in batches this occurs when the batch is executed. In batches, Cluster mode and standalone mode commands are the same with only a few exceptions for Select, Move, Scan, and PubSub commands. This is slightly different from regular contexts where some commands may have specific routing options that are available when using the ClusterClient.
+
+```go
+// In regular client context
+client := glide.NewClient(...)
+// Executes immediately with context
+client.Set(context.Background(), "key1", "value1")
+
+// In transaction context 
+tx := pipeline.NewStandaloneBatch(true)
+// Queued for later execution
+tx.Set("key1", "value1")
+// Execute transaction with context
+results, err := client.Exec(context.Background(), *tx, false)
+```
+
+### Optimistic Concurrency Support
+
+Valkey's optimistic locking mechanism is supported through:
+
+- Implementation of `Watch()` that monitors keys for changes,
+- Implementation of `Unwatch()` that releases monitoring,
+- Proper handling of transaction failure when watched keys change,
+- Passing watch state through to the Valkey server during execution.
```

Comment:
Is the persona reading this blog going to know what this means. 

I would also remove "Proper handling of transaction failure when watched keys change", since I don't know what that means.